src/inbox/mod_inbox.erl

```diff
@@ -209,11 +210,24 @@ process_iq(Acc, From, _To, #iq{type = set, sub_el = QueryEl} = IQ, _Extra) ->
             {Acc, Res}
     end.
 
+-spec with_rsm([inbox_res()], get_inbox_params()) -> [inbox_res()].
+with_rsm(List, #{order := asc, filter_on_jid := BinJid, rsm := #rsm_in{}}) ->
+    lists:reverse(drop_filter_on_jid(List, BinJid, List));
 with_rsm(List, #{order := asc, rsm := #rsm_in{}}) ->
     lists:reverse(List);
+with_rsm(List, #{order := desc, filter_on_jid := BinJid}) ->
+    drop_filter_on_jid(List, BinJid, List);
 with_rsm(List, _) ->
     List.
 
+-spec drop_filter_on_jid([inbox_res()], binary(), [inbox_res()]) -> [inbox_res()].
```

Comment:
This function can skip multiple results until it gets to the jid in question. This means that the number of returned messages would be smaller than the requested `max` value (?)
Also, why would we even care about the jid being somewhere in the middle of the results?

_"The first item in the page returned by the responding entity MUST be the item that immediately follows the item that the requesting entity indicated in the <after/> element"_

If the jid matches, but the timestamp doesn't, then the user just provided a wrong ID, right? Why try to fix it here?

Reply:
My concern is that there can be for example three conversations with the same timestamp, so we don't know in which order they will be returned and if any of them was the one that was returned in the previous page. So if a page returns for example the first two of these three conversations at the end, and use that second as an ID, when requesting with that ID we will get all three of these conversations first, and we need to discard the first two 😕